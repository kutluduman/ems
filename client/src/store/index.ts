import { configureStore, ThunkAction, Action } from "@reduxjs/toolkit";
import auth from "./slices/auth";
import employee from "./slices/employee";

export const store = configureStore({
  reducer: {
    auth,
    employee,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;