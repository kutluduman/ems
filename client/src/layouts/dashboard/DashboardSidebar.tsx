import { useEffect } from "react";
import { Link as RouterLink, useLocation } from "react-router-dom";
import { styled } from "@mui/material/styles";
import { Box, Link, Drawer, Typography, Avatar } from "@mui/material";

import Logo from "../../components/Logo";
import Scrollbar from "../../components/ScrollBar";
import NavSection from "../../components/NavSection";
import { MHidden, Width } from "../../components/@material-extend";
import sidebarConfig from "./SidebarConfig";
import { selectEmployee } from "../../store/slices/auth";
import { useAppSelector } from "../../store/hooks";

const DRAWER_WIDTH = 280;

const RootStyle = styled("div")(({ theme }) => ({
  [theme.breakpoints.up("lg")]: {
    flexShrink: 0,
    width: DRAWER_WIDTH,
  },
}));

const AccountStyle = styled("div")(({ theme }: { [key: string]: any }) => ({
  display: "flex",
  alignItems: "center",
  padding: theme.spacing(2, 2.5),
  borderRadius: theme.shape.borderRadiusSm,
  backgroundColor: theme.palette.grey[200],
}));

interface DashboardSidebarProps {
  isOpenSidebar: boolean;
  onCloseSidebar: (
    event?: {},
    reason?: "backdropClick" | "escapeKeyDown"
  ) => void;
}

const DashboardSidebar = ({
  isOpenSidebar,
  onCloseSidebar,
}: DashboardSidebarProps) => {
  const { pathname } = useLocation();
  const employee = useAppSelector(selectEmployee);

  useEffect(() => {
    if (isOpenSidebar) {
      onCloseSidebar();
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [pathname]);

  const renderContent = (
    <Scrollbar
      sx={{
        height: "100%",
        "& .simplebar-content": {
          height: "100%",
          display: "flex",
          flexDirection: "column",
        },
      }}
    >
      <Box sx={{ px: 2.5, py: 3 }}>
        <Box component={RouterLink} to="/" sx={{ display: "inline-flex" }}>
          <Logo />
        </Box>
      </Box>

      <Box sx={{ mb: 5, mx: 2.5 }}>
        <Link underline="none" component={RouterLink} to="#">
          <AccountStyle>
            <Avatar
              src="https://minimal-kit-react.vercel.app/static/mock-images/avatars/avatar_default.jpg"
              alt="photoURL"
            />
            <Box sx={{ ml: 2 }}>
              <Typography variant="subtitle2" sx={{ color: "text.primary" }}>
                {employee?.name} {employee?.surname}
              </Typography>
              <Typography variant="body2" sx={{ color: "text.secondary" }}>
                {employee?.title}
              </Typography>
            </Box>
          </AccountStyle>
        </Link>
      </Box>

      <NavSection navConfig={sidebarConfig} />
    </Scrollbar>
  );

  return (
    <RootStyle>
      <MHidden width={Width.lgUp}>
        <Drawer
          open={isOpenSidebar}
          onClose={onCloseSidebar}
          PaperProps={{
            sx: { width: DRAWER_WIDTH },
          }}
        >
          {renderContent}
        </Drawer>
      </MHidden>

      <MHidden width={Width.lgDown}>
        <Drawer
          open
          variant="persistent"
          PaperProps={{
            sx: {
              width: DRAWER_WIDTH,
              bgcolor: "background.default",
            },
          }}
        >
          {renderContent}
        </Drawer>
      </MHidden>
    </RootStyle>
  );
}

export default DashboardSidebar;
