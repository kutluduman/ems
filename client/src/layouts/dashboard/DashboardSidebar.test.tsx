import { create } from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";

import ThemeConfig from "../../theme";
import { store } from "../../store";
import DashboardSidebar from "./DashboardSidebar";

describe("DashboardSidebar Component", () => {
  test("renders the DashboardSidebar Component", () => {
    const tree = create(
      <Provider store={store}>
        <Router>
          <ThemeConfig>
            <DashboardSidebar isOpenSidebar={false} onCloseSidebar={() => {}} />
          </ThemeConfig>
        </Router>
      </Provider>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});