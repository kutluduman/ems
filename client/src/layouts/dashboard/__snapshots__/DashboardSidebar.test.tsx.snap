// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardSidebar Component renders the DashboardSidebar Component 1`] = `
<div
  className="css-1y6sui9"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-6lzitz-MuiPaper-root-MuiDrawer-paper"
      style={
        Object {
          "visibility": undefined,
        }
      }
    >
      <div
        className="css-jjtu05"
      >
        <div
          className="css-1i9qb4p"
          data-simplebar={true}
        >
          <div
            className="simplebar-wrapper"
          >
            <div
              className="simplebar-height-auto-observer-wrapper"
            >
              <div
                className="simplebar-height-auto-observer"
              />
            </div>
            <div
              className="simplebar-mask"
            >
              <div
                className="simplebar-offset"
              >
                <div
                  className="simplebar-content-wrapper"
                >
                  <div
                    className="simplebar-content"
                  >
                    <div
                      className="MuiBox-root css-1dk38ps"
                    >
                      <a
                        className="MuiBox-root css-vxcmzt"
                        href="/"
                        onClick={[Function]}
                        sx={
                          Object {
                            "display": "inline-flex",
                          }
                        }
                      >
                        <img
                          className="MuiBox-root css-1ngw63v"
                          src="https://minimal-kit-react.vercel.app/static/logo.svg"
                        />
                      </a>
                    </div>
                    <div
                      className="MuiBox-root css-1jliix"
                    >
                      <a
                        className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1i6eu10-MuiTypography-root-MuiLink-root"
                        href="/"
                        onBlur={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                      >
                        <div
                          className="css-x858a4"
                        >
                          <div
                            className="MuiAvatar-root MuiAvatar-circular css-1kgdhtw-MuiAvatar-root"
                          >
                            <img
                              alt="photoURL"
                              className="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                              src="https://minimal-kit-react.vercel.app/static/mock-images/avatars/avatar_default.jpg"
                            />
                          </div>
                          <div
                            className="MuiBox-root css-d0uhtl"
                          >
                            <h6
                              className="MuiTypography-root MuiTypography-subtitle2 css-1pyztyk-MuiTypography-root"
                            >
                               
                            </h6>
                            <p
                              className="MuiTypography-root MuiTypography-body2 css-1aaagy3-MuiTypography-root"
                            />
                          </div>
                        </div>
                      </a>
                    </div>
                    <div
                      className="MuiBox-root css-0"
                    >
                      <ul
                        className="MuiList-root css-1mk9mw3-MuiList-root"
                      >
                        <a
                          className="MuiListItemButton-root MuiButtonBase-root css-7oyo6l-MuiButtonBase-root-MuiListItemButton-root"
                          href="/dashboard/employees"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                        >
                          <div
                            className="MuiListItemIcon-root css-1gxhmaf-MuiListItemIcon-root"
                          >
                            <svg
                              aria-hidden={true}
                              className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                              data-testid="PeopleAltIcon"
                              focusable="false"
                              viewBox="0 0 24 24"
                            >
                              <path
                                d="M16.67 13.13C18.04 14.06 19 15.32 19 17v3h4v-3c0-2.18-3.57-3.47-6.33-3.87z"
                                fillRule="evenodd"
                              />
                              <circle
                                cx="9"
                                cy="8"
                                fillRule="evenodd"
                                r="4"
                              />
                              <path
                                d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4c-.47 0-.91.1-1.33.24C14.5 5.27 15 6.58 15 8s-.5 2.73-1.33 3.76c.42.14.86.24 1.33.24zm-6 1c-2.67 0-8 1.34-8 4v3h16v-3c0-2.66-5.33-4-8-4z"
                                fillRule="evenodd"
                              />
                            </svg>
                          </div>
                          <div
                            className="MuiListItemText-root css-vt7cuc-MuiListItemText-root"
                          >
                            employees
                          </div>
                        </a>
                      </ul>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="simplebar-placeholder"
            />
          </div>
          <div
            className="simplebar-track simplebar-horizontal"
          >
            <div
              className="simplebar-scrollbar"
            />
          </div>
          <div
            className="simplebar-track simplebar-vertical"
          >
            <div
              className="simplebar-scrollbar"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
